src/katello/tracer/dnf.py

```diff
@@ -0,0 +1,66 @@
+import re
+import os
+import subprocess
+from katello.tracer.SystemdDbus import SystemdDbus
+
+class DnfTracerApp:
+    def __init__(self, name, helper, type):
+        self.name = name
+        self.helper = helper
+        self.type = type
+
+
+def get_name(pid):
+    bus = SystemdDbus()
+    if pid and bus.unit_path_from_pid(pid):
+        if not bus.has_service_property_from_pid(pid, 'PAMName'):
+            Id = bus.get_unit_property_from_pid(pid, 'Id')
+            if Id and re.search("\.service$", Id):
+                return re.sub('\.service$', '', Id)
+    return None
+
+
+def run_command(command):
+    os.environ['LANG'] = 'C'
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+   
+    return_output = ''
+    return_code = process.wait()
+    
+    if return_code == 0:
+        output = process.stdout.read().decode('utf-8')
+        return_output = output.strip()
+    return return_code, return_output
+
+
+def collect_services_state():
+    command = "dnf needs-restarting"
+    retval, output = run_command(command)
+
+    lines = output.split('\n')
+    pids = [line.split(' : ')[0].strip() for line in lines if ' : ' in line]
+
+    apps = set()
+    for pid in pids:
+        service = get_name(pid)
+        if service is not None:
+            app = DnfTracerApp(service, "systemctl restart " + service, "systemd")
```

Comment:
I'm seeing app types "systemd" and "static" but not "session" or "daemon".   It looks like the "systemctl" type introduced here is equivalent to the old "daemon" types.

 The session types tell you if you need to log out and log back in to see changes. Is that not going to be possible with `dnf needs-restarting`?  If not we might need to fall back to python3-tracer for that information, or I suppose pull more code from them like what was done with `SystemdDbus()`

![image (7)](https://github.com/Katello/katello-host-tools/assets/14796566/f3ccd14a-1cd8-478d-8b21-d442af18c700)

Reply:
I will have a look at the session type. 
Just another question to the session type: when is it necessary to log-in / log-out to "restart" a session? If the tracer information is collected via REX, I thought you always have a new session and no restart should be necessary?

Reply:
Tracer information is uploaded to Katello via a dnf plugin. REX is used only to execute the helpers.

Reply:
Added session type.